src/els_elvis_diagnostics.erl

```diff
@@ -0,0 +1,83 @@
+%%==============================================================================
+%% Compiler diagnostics
+%%==============================================================================
+-module(els_elvis_diagnostics).
+
+%%==============================================================================
+%% Behaviours
+%%==============================================================================
+-behaviour(els_diagnostics).
+
+%%==============================================================================
+%% Exports
+%%==============================================================================
+-export([ diagnostics/1 ]).
+
+%%==============================================================================
+%% Includes
+%%==============================================================================
+-include("erlang_ls.hrl").
+
+%%==============================================================================
+%% Callback Functions
+%%==============================================================================
+-spec diagnostics(uri()) -> [diagnostic()].
+diagnostics(Uri) ->
+  RelFile = els_utils:project_relative(Uri),
+
+  %% Note: elvis_core:rock_this requires a file path relative to the
+  %%       project root, formatted as a string.
+  WS = try elvis_core:rock_this(RelFile)
+       catch Err ->
+           lager:info("Elvis error.[Err=~p] ", [Err]),
+           ok  %% TODO: return a diagnostic saying elvis failed?
```

Comment:
Regarding the TODO: I would log this as a `lager:warning` or `lager:error` and call it a day for now.